islands/src20/SRC20Card.tsx

```diff
@@ -0,0 +1,160 @@
+import { useState } from "preact/hooks";
+import { SRC20Row } from "globals";
+import { convertToEmoji } from "$lib/utils/util.ts";
+
+interface SRC20CardProps {
+  src20: SRC20Row;
+  variant?: "deploy" | "trending" | "minting";
+  onImageClick?: (imgSrc: string) => void;
+}
+
+export function SRC20Card(
+  { src20, variant = "deploy", onImageClick }: SRC20CardProps,
+) {
+  const [isHovered, setIsHovered] = useState(false);
+
+  const href = `/src20/${encodeURIComponent(convertToEmoji(src20.tick))}`;
+  const mintHref = `/stamping/src20/mint?tick=${
+    encodeURIComponent(src20.tick)
+  }&trxType=olga`;
+  const progress = src20.progress || "0";
+  const progressWidth = `${progress}%`;
+
+  const handleMintClick = () => {
+    globalThis.location.href = mintHref;
+  };
+
+  const imageUrl = src20.stamp_url ||
+    src20.deploy_img ||
+    `/content/${src20.tx_hash}.svg` ||
+    `/content/${src20.deploy_tx}`;
+
+  return (
+    <div
+      class="flex text-sm justify-between items-center rounded-md hover:border-[#9900EE] hover:shadow-[0px_0px_20px_#9900EE] w-full bg-gradient-to-br from-[#0A000F00] via-[#14001FFF] to-[#1F002EFF]"
+      onMouseEnter={() => setIsHovered(true)}
+      onMouseLeave={() => setIsHovered(false)}
+    >
+      {/* Left Section - Image and Title */}
+      <div class="p-3 uppercase cursor-pointer flex gap-6">
+        <img
+          src={imageUrl}
+          class="w-[65px] h-[65px]"
+          onClick={() => onImageClick?.(imageUrl)}
+          alt={src20.tick}
+        />
+        <div class="flex flex-col justify-between">
+          <a
+            href={href}
+            class={`text-2xl font-bold ${
+              isHovered ? "text-[#AA00FF]" : "text-[#666666]"
+            } flex gap-4`}
+          >
+            {convertToEmoji(src20.tick)}
```

Comment:
each social icons should only display if  src20.email !=null src20.web != null, etc.

Reply:
👍